ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/transaction/jts/jca/XATerminatorImple.java

```diff
@@ -452,7 +452,13 @@ public boolean beforeCompletion (Xid xid) throws javax.transaction.SystemExcepti
 
     public Transaction getTransaction(Xid xid) throws XAException {
         // first see if the xid is a root coordinator
-        return com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionImple.getTransaction(new XidImple(xid).getTransactionUid());
+        Transaction transaction = com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionImple
```

Comment:
@mmusgrov it looks like this is where getTransactionUid was introduced to JTS side: https://github.com/jbosstm/narayana/commit/9f47df920514255c612c9d9c1cb9c7ad5ad5963a#diff-938c9dfac82d12a192ec11ae14cacf09